sklearn_extra/cluster/_k_medoids.py

```diff
@@ -493,3 +510,255 @@ def _kpp_init(self, D, n_clusters, random_state_, n_local_trials=None):
+                            replace=False,
+                        ),
+                    ]
+                )
+            pam = KMedoids(
+                n_clusters=self.n_clusters,
+                metric=self.metric,
+                method="pam",
+                init=self.init,
+                max_iter=self.max_iter,
+                random_state=random_state_,
+            )
+            pam.fit(X[sample_idxs])
+            self.cluster_centers_ = pam.cluster_centers_
+            self.inertia_ = self._compute_inertia(self.transform(X))
+
+            if pam.inertia_ < best_score:
+                best_score = self.inertia_
+                medoids_idxs = pam.medoid_indices_
+                best_sample_idxs = sample_idxs
+
+        self.medoid_indices_ = medoids_idxs
+        self.labels_ = np.argmin(self.transform(X), axis=1)
+        self.n_iter_ = self.samples
+
+        return self
+
+    def _compute_inertia(self, distances):
+        """Compute inertia of new samples. Inertia is defined as the sum of the
+        sample distances to closest cluster centers.
+
+        Parameters
+        ----------
+        distances : {array-like, sparse matrix}, shape=(n_samples, n_clusters)
+            Distances to cluster centers.
+
+        Returns
+        -------
+        Sum of sample distances to closest cluster centers.
+        """
+
+        # Define inertia as the sum of the sample-distances
+        # to closest cluster centers
+        inertia = np.sum(np.min(distances, axis=1))
+
+        return inertia
+
+    def transform(self, X):
+        """Transforms X to cluster-distance space.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_query, n_features), \
+                or (n_query, n_indexed) if metric == 'precomputed'
+            Data to transform.
+
+        Returns
+        -------
+        X_new : {array-like, sparse matrix}, shape=(n_query, n_clusters)
+            X transformed in the new space of distances to cluster centers.
+        """
+        X = check_array(X, accept_sparse=["csr", "csc"])
+
+        if self.metric == "precomputed":
+            check_is_fitted(self, "medoid_indices_")
+            return X[:, self.medoid_indices_]
+        else:
+            check_is_fitted(self, "cluster_centers_")
+
+            Y = self.cluster_centers_
+            return pairwise_distances(X, Y=Y, metric=self.metric)
+
+    def predict(self, X):
+        """Predict the closest cluster for each sample in X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_query, n_features), \
+                or (n_query, n_indexed) if metric == 'precomputed'
+            New data to predict.
+
+        Returns
+        -------
+        labels : array, shape = (n_query,)
+            Index of the cluster each sample belongs to.
+        """
+        X = check_array(X, accept_sparse=["csr", "csc"])
```

Comment:
Should also work,

```suggestion
        X = check_array(X, accept_sparse=["csr", "csc"], dtypes=[np.float64, np.float32])
```
but we would need to add tests for it.